addons/udes_stock/wizard/change_quant_location.py

```diff
@@ -1,55 +1,120 @@
+        user_allowed = self.env.user.has_group("udes_stock.group_manage_reserved_packages")
+        warehouse = self.env.user.get_user_warehouse()
+        return warehouse.u_allow_create_picking_reserved_package and user_allowed
 
     @api.multi
     def create_picking(self):
+        Picking = self.env["stock.picking"]
+        Package = self.env["stock.quant.package"]
+
         self.ensure_one()
-        Picking = self.env['stock.picking']
-        Package = self.env['stock.quant.package']
+        packages = Package.browse(self.env.context["active_ids"])
+        quants = packages._get_contained_quants()
+
+        # Check if we are allowed to move the reserved quants
+        total_quantity_reserved = sum(quants.mapped("reserved_quantity"))
+        if total_quantity_reserved:
+            if not self._check_reserved_packages_permitted():
+                # If not allowed to move reserved packages use existing functionality
+                # to show full details on error
+                quants.assert_not_reserved()
+            total_quantity = sum(quants.mapped("quantity"))
+            if total_quantity != total_quantity_reserved:
+                # If all quants are not fully reserved then error
+                raise ValidationError(
+                    _("Packages/Pallets must be either entirely reserved or unreserved.")
+                )
 
-        packages = Package.browse(self.env.context['active_ids'])
-        quant_ids = packages._get_contained_quants().ids
-        location = packages.mapped('location_id')
+        location = packages.mapped("location_id")
         if len(location) == 1:
             location_id = location.id
         else:
             location_id = self._default_stock()
         params = {
-            'quant_ids': quant_ids,
-            'location_id': location_id,
-            'picking_type_id': self.picking_type_id.id,
+            "location_id": location_id,
+            "picking_type_id": self.picking_type_id.id,
+            "location_dest_id": (
+                self.location_dest_id.id or self.picking_type_id.default_location_dest_id.id
+            ),
         }
         if self.reference:
-            params['origin'] = self.reference
+            params["origin"] = self.reference
 
-        new_picking = Picking.create_picking(**params)
-        # the picking is created with default location_dest_id
-        # of the picking type so odoo creates the route if needed
-        # after location_dest_id is updated to the specific
-        # location set at self.location_dest_id
-        if self.location_dest_id:
-            new_picking.location_dest_id = self.location_dest_id
+        # If quants are reserved move them to a new picking
+        if total_quantity_reserved:
+            # Create a new picking
+            new_picking = Picking.create(params)
+            movelines = packages.get_move_lines()
+            movelines = movelines.filtered(lambda ml: ml.picking_id.state == "assigned")
```

Comment:
`get_move_lines()` supports extra domain, so you can filter them out in the previous call.

Reply:
Added this in